### PR TITLE
Local auth

### LINKED ISSUE #6: Auth v2

### PR DESCRIPTION
**Pull request by ZuluNovember**
Closes #6 

# Implement local sign-in strategy
- Implemented local authentication on `auth/signin` endpoint.
- It uses common callback and signout controllers with google auth.

### CODE REVIEW

**Review by Kishimoto96 — COMMENTED** [review]
Thanks for the great work, do we need a signup function also in the routes or are we just gonna use google auth for signing up?

**Review by Kishimoto96 — APPROVED** [review]
perfect thank you so much for your consideration

File: src/utils/localAuth.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      if (!isCorrectPw) {
+        return callback(null, false, {
+          message: 'Incorrect username or password',
+        });
+      }
+
+      return callback(null, user);
+    } catch (err) {
+      return logger.error(err);
+    }
```
Comment (Shrreya) [inline src/utils/localAuth.js:27]:
```suggestion
    } catch (err) {
      logger.error(err);
      return callback(null, false, err);
    }
```

**Review by Shrreya — CHANGES_REQUESTED** [review]
Just 1 small change request, rest looks good!

**Review by Shrreya — APPROVED** [review]

**Pull request merged**